pkg/compiler/import_test.go

```diff
@@ -61,3 +65,27 @@ func TestImportNameSameAsOwn(t *testing.T) {
 	}`
 	eval(t, src, big.NewInt(3))
 }
+
+func TestImportCycleIndirect(t *testing.T) {
+	src := `
+		package some
+		import "github.com/nspcc-dev/neo-go/pkg/compiler/testdata/importcycle/pkg2"
+		func Main() int {
+			return pkg2.A
+		}
+	`
+	_, _, err := compiler.CompileWithOptions("some.go", strings.NewReader(src), nil)
+	require.Error(t, err)
+}
+
+func TestImportCycleDirect(t *testing.T) {
```

Comment:
And this one is `Indirect` (pkg1 imports pkg2, and import cycle is in pkg2-pkg3).

But it's a nitpicking, tests are good.